src/aurora/lua/scriptman.cpp

```diff
@@ -0,0 +1,455 @@
+/* xoreos - A reimplementation of BioWare's Aurora engine
+ *
+ * xoreos is the legal property of its developers, whose names
+ * can be found in the AUTHORS file distributed with this source
+ * distribution.
+ *
+ * xoreos is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 3
+ * of the License, or (at your option) any later version.
+ *
+ * xoreos is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with xoreos. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/** @file
+ *  Lua script manager.
+ */
+
+extern "C" {
+
+#include "lua/lualib.h"
+
+#include "toluapp/tolua++.h"
+
+}
+
+#include "src/common/error.h"
+#include "src/common/util.h"
+
+#include "src/aurora/resman.h"
+#include "src/aurora/util.h"
+
+#include "src/aurora/lua/scriptman.h"
+#include "src/aurora/lua/stack.h"
+#include "src/aurora/lua/stackguard.h"
+#include "src/aurora/lua/variable.h"
+#include "src/aurora/lua/table.h"
+#include "src/aurora/lua/function.h"
+
+DECLARE_SINGLETON(Aurora::Lua::ScriptManager)
+
+namespace Aurora {
+
+namespace Lua {
+
+ScriptManager::ScriptManager() : _luaState(0), _regNestingLevel(0) {
+
+}
+
+ScriptManager::~ScriptManager() {
+	closeLuaState();
+
+	if (!_objectLuaInstances.empty()) {
+		warning("Lua instances were not freed properly");
+	}
+}
+
+void ScriptManager::init() {
+	assert(!ready());
+
+	openLuaState();
+	registerDefaultBindings();
+	executeDefaultCode();
+}
+
+void ScriptManager::deinit() {
+	assert(ready());
+
+	closeLuaState();
+}
+
+bool ScriptManager::ready() const {
+	return _luaState != 0;
+}
+
+void ScriptManager::executeFile(const Common::UString &path) {
+	assert(_luaState && _regNestingLevel == 0);
+
+	if (isIgnoredFile(path)) {
+		return;
+	}
+
+	Common::SeekableReadStream *stream = ResMan.getResource(path, kFileTypeLUC);
+	if (!stream) {
+		throw Common::Exception("No such LUC \"%s\"", TypeMan.setFileType(path, kFileTypeLUC).c_str());
+	}
+
+	Common::MemoryReadStream *memStream = stream->readStream(stream->size());
+	const char *data = reinterpret_cast<const char *>(memStream->getData());
+	const int dataSize = memStream->size();
+
+	const int execResult = lua_dobuffer(_luaState, data, dataSize, path.c_str());
+	if (execResult != 0) {
+		delete memStream;
```

Comment:
Using `boost::scoped_ptr` in other places is also on my TODO list (*). This is a good place as any to start with that, yes.

(*) Since currently xoreos-tools is without boost, I've been holding that off. I'm still unsure on whether I should just add boost to the xoreos-tools dependencies or copy the boost pointers templates into xoreos and/or xoreos-tools. To be fair, the major reason for not adding boost to xoreos-tools is the slowness of boost.m4, which could maybe be changed/fixed/handled instead...